sage/example-transition/src/rules/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::asset;
+
+use ajuna_primitives::asset_manager::{AssetInspector, AssetManager};
+use sage_api::Error;
+
+use frame_support::{
+	ensure,
+	pallet_prelude::{Decode, Encode, MaxEncodedLen, TypeInfo},
+};
+use parity_scale_codec::Codec;
+use sp_runtime::traits::{BlockNumber as BlockNumberT, Member};
+use std::marker::PhantomData;
+
+pub mod hero_jam;
+
+pub trait RuleVerifier {
+	type AccountId;
+	type AssetId;
+	fn verify(
+		&self,
+		account_id: &Self::AccountId,
+		asset_ids: &[Self::AssetId],
+	) -> Result<(), Error>;
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub enum Rule<AccountId, TransitionRuleset, AssetHandler> {
+	Not(Box<Self>),
+	Or(Box<Self>, Box<Self>),
+	And(Box<Self>, Box<Self>),
+	AssetCount(u8),
+	IsOwnerOf(AccountId, PhantomData<AssetHandler>),
+	Transition(TransitionRuleset),
+}
```

Comment:
I wonder if this enum is actually beneficial. I mean, we are maybe trying to reduce friction for non-rust devs, but we are talking to devs at least. I could imagine that it is actually less complicated if we just offer functions instead of introducing our enum here.